server/events.go

```diff
@@ -618,13 +623,17 @@ func (s *Server) initEventTracking() {
 		s.Errorf("Error setting up internal latency tracking: %v", err)
 	}
 
-	// These are for system account exports for debugging from client applications.
-	sacc := s.sys.account
-
 	// This is for simple debugging of number of subscribers that exist in the system.
 	if _, err := s.sysSubscribeInternal(accSubsSubj, s.debugSubscribers); err != nil {
 		s.Errorf("Error setting up internal debug service for subscribers: %v", err)
 	}
+}
+
+// add all exports a system account will need
+func (s *Server) addSystemAccountExports(sacc *Account) {
```

Comment:
Since we call eventsEnabled(), caller needs to have s.mu.Lock(). If that is the case, maybe add comment:
```
// Server lock held on entry
```
at top of this function.

Reply:
will add

Reply:
Actually this was a result of moving the code out of initEventTracking, no lock held there either. I'll change it to EventsEnabled